app/RecordCollectors/Vatify.php

```diff
@@ -0,0 +1,356 @@
+		$this->getBearerToken();
+		if (empty($this->bearerToken)){
+			$this->response['error'] = \App\Language::translate('LBL_VATIFY_NO_AUTH', 'Other.RecordCollector');
+		}
+		$params['country'] = $country;
+		$params['identifier'] = $vatNumber;
+
+		$this->getDataFromApi($params);
+		$this->loadData();
+		return $this->response;
+	}
+
+	/**
+	 * Function fetching company data by VAT ID and Country.
+	 *
+	 * @param array $params
+	 *
+	 * @return void
+	 */
+	private function getDataFromApi(array $params): void
+	{
+		$response = [];
+		$link = '';
+		try {
+			$response = \App\Json::decode(\App\RequestHttp::getClient()->get($this->url  . 'query?' . http_build_query($params), ['headers' => ['Authorization' => 'Bearer '. $this->bearerToken]])->getBody()->getContents());
+		} catch (\GuzzleHttp\Exception\ClientException $e) {
+			\App\Log::warning($e->getMessage(), 'RecordCollectors');
+			$this->response['error'] = $e->getResponse()->getReasonPhrase();
+			return;
+		}
+
+		if ('IN_PROGRESS' === $response['query']['status']) {
+			$link = $response['query']['links'][0]['href'];
+		} else {
+			return;
+		}
+
+		try {
+			$response = \App\Json::decode(\App\RequestHttp::getClient()->get($link)->getBody()->getContents());
+		} catch (\GuzzleHttp\Exception\ClientException $e) {
+			\App\Log::warning($e->getMessage(), 'RecordCollectors');
+			$this->response['error'] = $e->getResponse()->getReasonPhrase();
+			return;
+		}
+
+		if ('FINISHED' === $response['query']['status']){
+			$this->data = $this->parseData($response['result']['items'][0]['data']);
+		} else {
+			$this->data = [];
+		}
+	}
+
+	/**
+	 * Function parsing data to fields from API.
+	 *
+	 * @param array $data
+	 *
+	 * @return array
+	 */
+	private function parseData(array $data): array
+	{
+		return \App\Utils::flattenKeys($data, 'ucfirst');
+	}
+
+	/**
+	 * Function setup Api Key.
+	 *
+	 * @return void
+	 */
+	private function setCredentials(): void
+	{
+		if (($params = $this->getParams()) && !empty($params['client_id'] && !empty($params['access_key']))) {
+			$this->clientId = $params['client_id'];
+			$this->accessKey = $params['access_key'];
+		} else {
+			throw new \App\Exceptions\IllegalValue('You must fist setup Api Key in Config Panel', 403);
+		}
+	}
+
+	private function getBearerToken()
+	{
+		$credentials = base64_encode($this->clientId . ':' . $this->accessKey);
+		$options = [
+			'headers' => [
+				'Authorization' => 'Basic ' . $credentials
+			],
+			'grant_type' => 'client_credentials'
+		];
+
+		try {
+			$response = \App\Json::decode(\App\RequestHttp::getClient()->get('https://api.vatify.eu/v1/oauth2/token', $options)->getBody()->getContents());
+		} catch (\GuzzleHttp\Exception\ClientException $e) {
+			\App\Log::warning($e->getMessage(), 'RecordCollectors');
+			$this->response['error'] = $e->getResponse()->getReasonPhrase();
+			return;
+		}
+
+		if ($response['access_token']) {
+			$this->bearerToken = $response['access_token'];
+		}
```

Comment:
if (empty($this->bearerToken)){
			$this->response['error'] = \App\Language::translate('LBL_VATIFY_NO_AUTH', 'Other.RecordCollector');
		}